tests/testthat/test-sexp.R

```diff
@@ -0,0 +1,143 @@
+context("sexp")
+
+test_that("Can use parens", {
+  expr <- list("*", "a", list("+", "b", "c"))
+  expect_equal(
+    generate_dust_sexp(expr, NULL, NULL),
+    "a * b + c")
+  expr <- list("*", "a", list("(", list("+", "b", "c")))
+  expect_equal(
+    generate_dust_sexp(expr, NULL, NULL),
+    "a * (b + c)")
+})
+
+
+test_that("^ becomes std::pow", {
+  expr <- list("^", "a", "b")
+  expect_equal(
+    generate_dust_sexp(expr, NULL, NULL),
+    "std::pow(a, b)")
```

Comment:
Almost everything in odin is a floating point number of some type, so it's very unlikely that we'd use the `int` version. However, the compiler will be able to see if it can be used surely?

Reply:
This comment is very stream-of-conciousness and can probably be ignored... I was thinking of a possible C++11 differences, but I'm sure the compiler will work it out

Reply:
The >=0 check is not done, no - there are a huge number of things that like this that can cause problems but aren't checked (sqrt and log of negative numbers etc). This is all UB as far as odin is concerned 😁